source/vibe/core/core.d

```diff
@@ -877,18 +892,64 @@ unittest {
 
 
 /**
-	Returns a new armed timer.
-
-	Note that timers can only work if an event loop is running, explicitly or
-	implicitly by running a blocking operation, such as `sleep` or `File.read`.
+	Creates a new timer, that will fire `callback` after `timeout`
+
+    There are two kinds of timer: one-off timers and periodic timers.
+    One-off timers fire only once, after a specific amount of time,
+    while periodic timer fire at a regular interval.
+
+    One-off_timers:
+    One-off timers can be used for performing a task after a specific delay,
+    or for the initial setup of an application. For example, the following
+    is a common pattern for a Vibe.d application:
+    ---
+    import vibe.d;
+
+    /// Setup the application and let other server know we are running
+    void setup (Config conf) {
+        // Magic!
+    }
+
+    int main (string[] args) {
+        Config conf = parseArguments(&args);
+        setTimer(0.msecs, () => setup(conf));
+        return runApplication(&args);
+    }
+    ---
+
+    As timers only run when the event loop is running, and `runApplication`
+    will start the event loop, `setupTimer` will just register the closure
+    that calls `setup` to be executed as soon as the event loop is ready.
+
+    Periodic_timers:
+    Periodic timers will trigger for the first time after `timeout`,
+    then at best every `timeout` period after this. Periodic timers may be
+    explicitly stopped by calling the `Timer.stop()` method on the return value
+    of this function.
+
+    As timer are non-preemtive (see the "Preemption" section), user code does
+    not need to compensate for time drift, as the time spent in the function
+    will not affect the frequency, unless the function takes longer to complete
+    than the timer.
+
+    Preemption:
+    Like other events in Vibe.d, timers are non-preemptive, meaning that
+    the currently executing function will not be interrupted to let a timer run.
+    This is usually not a problem in server applications, as any blocking code
+    will be easily noticed (the server will stop to handle requests), but might
+    come at a surprise in code that doesn't handle request.
+    If this is a problem, the solution is usually to either explicitly give
+    control to the event loop (by calling `yield`) or ensuring operations are
+    asynchronous (e.g. call functions from `vibe.core.file` instead of `std.file`).
 
 	Params:
 		timeout = Determines the minimum amount of time that elapses before the timer fires.
-		callback = If non-`null`, this delegate will be called when the timer fires
+		callback = A delegate to be called when the timer fires. Can be `null`,
+                   in which case the timer will not do anything.
```

Comment:
Might make sense to mention the possibility to use `wait()` on a timer with no callback set here.